### PR TITLE
Add infix modifier based on changes in 3.0.0-M2

### PR DESCRIPTION
**Pull request by tgodzik**
While working on infix I figured it would be nice to rework soft keywords a bit, since there was a bit of duplicated code with `&& .alllow` and this change will make them look similar to KwGiven etc.

I split the PR into two commits:
- first one only adds infix
- second one reworks soft keywords

Let me know what you think!

### CODE REVIEW

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala [hunk 1]
```diff
@@ -924,4 +924,81 @@ class MinorDottySuite extends BaseDottySuite {
       )
     )
   }
+
+  // infix modifier - can be appplied only to type and methods
+  test("infix-modifier") {
+
```
Comment (kpbochenek) [inline tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala:930]: it should work okay but I would just add a test for infix extension method

**Review by kpbochenek — CHANGES_REQUESTED** [review]
Cool refactoring of Soft keywords, really like a new approach! :+1: 

Just one blocking discrepancy with specs/repl

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala [hunk 2]
```diff
@@ -924,4 +924,81 @@ class MinorDottySuite extends BaseDottySuite {
...
+    )
+
+    runTestError[Stat](
+      """|infix val a = b
+         |""".stripMargin,
+      "infix not allowed here. Only defs or types can be infix"
+    )
+
+    runTestError[Stat](
+      """|infix class A
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala:997]: Thanks! I haven't read that. Might be easier to not check it on the parser level then.

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala [hunk 3]
```diff
@@ -924,4 +924,81 @@ class MinorDottySuite extends BaseDottySuite {
...
+         |""".stripMargin,
+      "infix not allowed here. Only defs or types can be infix"
+    )
+
```
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala]: Added a new test in a new test suite

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala (hunk 1 above)
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala]: Done!

File: tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (tgodzik) [reply tests/shared/src/test/scala/scala/meta/tests/parsers/dotty/MinorDottySuite.scala]: Disabled the check.

**Review by kpbochenek — APPROVED** [review]
LGTM! :+1:

**Pull request merged**